processing/src/main/java/io/druid/granularity/PeriodGranularity.java

```diff
@@ -306,7 +314,7 @@ private long truncateMillisPeriod(final long t)
   {
     // toStandardDuration assumes days are always 24h, and hours are always 60 minutes,
     // which may not always be the case, e.g if there are daylight saving changes.
-    if (chronology.days().isPrecise() && chronology.hours().isPrecise()) {
+    if (isPrecise()) {
```

Comment:
If all period used in PeriodGranualrity are composed with days and hours, it's ok. But if it's not (week, month, year), we should make it sure that all those fields are precise, too. And if it's just composed with seconds or minutes, we don't need to bother whatever days and hours are.